### PR TITLE
Add userinfo in reviews

### PR DESCRIPTION
**Pull request by pablo-desperados**
- created ability to display user photo, first name, and last name in reviews
- wrote tests for this ability

accomplished these things by:
- associating reviews in the show locations container with users
- including the desired props in the review tile
- creating a new function to ensure our new review displayed when posted in the show container
- added method in the user class model

### CODE REVIEW

File: app/controllers/api/v1/locations_controller.rb [hunk 1]
```diff
@@ -10,9 +10,10 @@ def index
 
   def show
     location = Location.find(params["id"])
-    reviews = location.reviews
-
-    payload = {"location":location, "reviews": reviews}
+    all_reviews = location.reviews
+    all_users = Location.findusers(all_reviews)
+    combined_arr = Location.combine(all_reviews, all_users)
+    payload = {"location":location, "reviews": combined_arr}
```
Comment (CassandraNewell) [inline app/controllers/api/v1/locations_controller.rb]: Can you clarify why this change was necessary?

File: app/javascript/react/containers/ShowLocationContainer.js [hunk 2]
```diff
@@ -57,13 +58,16 @@ class ShowLocationContainer extends React.Component {
...
 
-  componentDidMount(){
+  loadContent(){
     fetch(`/api/v1/locations/${this.props.match.params.id}`)
     .then(response => response.json())
     .then(response => {
       this.setState({chosenLocation: response.location, reviews: response.reviews})
     })
   }
+  componentDidMount(){
```
Comment (CassandraNewell) [inline app/javascript/react/containers/ShowLocationContainer.js]: Please add an empty line ⬆️

File: app/models/location.rb [hunk 3]
```diff
@@ -9,4 +9,22 @@ class Location < ApplicationRecord
 
   mount_uploader :location_picture, LocationPictureUploader
 
+  def self.findusers(reviews)
+    completed_hash = []
+    reviews.each do |review|
+      user = User.find(review.user_id)
+      completed_hash << user
+    end
+    return completed_hash
```
Comment (CassandraNewell) [inline app/models/location.rb]: Remember that you don't need final `return`s in Ruby!

File: app/models/location.rb [hunk 4]
```diff
@@ -9,4 +9,22 @@ class Location < ApplicationRecord
 
   mount_uploader :location_picture, LocationPictureUploader
 
+  def self.findusers(reviews)
+    completed_hash = []
+    reviews.each do |review|
```
Comment (CassandraNewell) [inline app/models/location.rb]: Replace with a `map` :)

File: app/controllers/api/v1/locations_controller.rb (hunk 1 above)
Comment (pablo-desperados) [reply app/controllers/api/v1/locations_controller.rb]: We needed to pass the information of the user who wrote the review along with the information of the review itself to react. Before the structure of payload wouldn't let us create several ReviewTiles with one map.

Now we combined the info of the review and the user info in one object.
